app/assets/javascripts/oxalis/model/volumetracing/volumelayer.js

```diff
@@ -35,29 +37,59 @@ export class VoxelIterator {
     width: number,
     height: number,
     minCoord2d: Vector2,
-    get3DCoordinate: Vector2 => Vector3 = () => [0, 0, 0],
+    get3DCoordinate?: Vector2 => Vector3 = () => [0, 0, 0],
+    boundingBox?: ?BoundingBox,
   ) {
     this.map = map;
     this.width = width;
     this.height = height;
     this.minCoord2d = minCoord2d;
     this.get3DCoordinate = get3DCoordinate;
-    if (!this.map[0][0]) {
+    this.boundingBox = boundingBox;
+    const firstCoordinate = this.get3DCoordinate([this.minCoord2d[0], this.minCoord2d[1]]);
+    if (!this.map[0][0] || !this.isCoordinateInBounds(firstCoordinate)) {
```

Comment:
Can you invert the if condition and switch the body and else block. In my opinion, that's easier to read. So, it would be something like:

```
if (map[0][0] != null && isInBounds...) {
  this.next = firstCoordinate;
} else {
  this.getNext();
}
```